src/jdk.jfr/share/classes/jdk/jfr/internal/instrument/FileChannelImplInstrumentor.java

```diff
@@ -67,136 +67,202 @@ public void force(boolean metaData) throws IOException {
     @JIInstrumentationMethod
     public int read(ByteBuffer dst) throws IOException {
         EventConfiguration eventConfiguration = EventConfigurations.FILE_READ;
-        if (!eventConfiguration.isEnabled()) {
```

Comment:
lets leave this 'early-out' in and extend the test to check that both events are disabled